File: sorting/bubble_sort.js
const a = Array.from({length : 100}, () => Math.floor(Math.random() * 100) + 1);
function swap(i, j) {
  let temp = a[i]
  a[i] = a[j]
  a[j] = temp
}

console.log(a)
function bubble_sort(a) {
  let len_a = a.length
  let steps = 0
  let swapped = true
  for (let i = 0; i < len_a; i++) {
    for (let j = 0; j < len_a - i - 1; j++) {
      if (a[j] > a[j + 1]) {
        swap(j, j + 1)
        swapped = true
        steps += 1
      }
    }
  }
  return steps
}
let steps = bubble_sort(a)
console.log(steps)
console.log(a)

File: sorting/bubble_sort.py
import random

    a = random.sample(range(1, 101), 100)

                          def swap(i, j) :a[i], a[j] = a[j], a[i]

                                                         def bubble_sort(a) :len_a = len(a) steps = 0 swapped = True while swapped:swapped = False for i in range(len_a - 1) : if a[i]> a[i + 1] :swap(i, i + 1) swapped = True steps += 1 return steps

                                                                                                                                                                                                       steps = bubble_sort(a) print(steps)
